pkg/api/v1alpha1/cluster.go

```diff
@@ -833,3 +834,11 @@ func validateMDUpgradeRolloutStrategy(w *WorkerNodeGroupConfiguration) error {
 
 	return nil
 }
+
+// IsRollingNotScale compares current CAPI objects and desired CAPI objects
+// returns true on rolling upgrade, false on scale upgrade.
+// should error if rolling and scale at the same time
+// todo: IsRollingNotScale
+func IsRollingNotScale(ctx context.Context, client client.Client, cluster *Cluster) (bool, error) {
```

Comment:
I see some commentary on this moving around. Whereever it lands, could we name it `IsRollingUpgrade`, there's really no need to describe both the affirmative and negative in a name and it generally makes it less usable elsewhere.